app/src/main/java/ch/epfl/sdp/utils/DependencyFactory.java

```diff
@@ -0,0 +1,77 @@
+package ch.epfl.sdp.utils;
+
+import androidx.annotation.VisibleForTesting;
+
+import ch.epfl.sdp.database.firebase.api.ClientDatabaseAPI;
+import ch.epfl.sdp.database.firebase.api.ClientFirestoreDatabaseAPI;
+import ch.epfl.sdp.database.firebase.api.CommonDatabaseAPI;
+import ch.epfl.sdp.database.firebase.api.CommonFirestoreDatabaseAPI;
+import ch.epfl.sdp.database.firebase.api.ServerDatabaseAPI;
+import ch.epfl.sdp.database.firebase.api.ServerFirestoreDatabaseAPI;
+import ch.epfl.sdp.login.AuthenticationAPI;
+import ch.epfl.sdp.login.FirebaseAuthenticationAPI;
+
+public class DependencyFactory {
+    private static AuthenticationAPI authenticationAPI;
+    private static CommonDatabaseAPI commonDatabaseAPI;
+    private static ServerDatabaseAPI serverDatabaseAPI;
+    private static ClientDatabaseAPI clientDatabaseAPI;
+    private static boolean testMode = false;
+
+    public static void setTestMode(boolean testMode) {
+        DependencyFactory.testMode = testMode;
+    }
+
+    public static boolean isTestMode() {
+        return testMode;
+    }
+
+    @VisibleForTesting
```

Comment:
I did not know this kind of things were possible, this literally changes everything!